### ISSUE TITLE
Using CustomAmiId when calling run_job_flow incorrectly throws ValidationException for all EMR versions between 5.10.0 & 5.31.0

### ISSUE THREAD

**Issue by PanJoseph** [spulec/moto#3474]
Right now the check in `run_job_flow` to see if a CustomAmiId is [valid](https://github.com/spulec/moto/blob/master/moto/emr/responses.py#L311) is done with default python string comparison instead of in a manner which would respect semantic versioning.
So `"emr-5.7.0"` is always considered greater than any value between `"5.1..."`(like 5.10.0) & `"5.6..."`(like 5.60.0) which is incorrect behavior

Reproducible like such:
```python
import boto3
from moto import mock_emr

run_job_flow_args = dict(
    Instances={
        "InstanceCount": 3,
        "KeepJobFlowAliveWhenNoSteps": True,
        "MasterInstanceType": "c3.medium",
        "Placement": {"AvailabilityZone": "us-east-1a"},
        "SlaveInstanceType": "c3.xlarge",
    },
    JobFlowRole="EMR_EC2_DefaultRole",
    LogUri="s3://mybucket/log",
    Name="cluster",
    ServiceRole="EMR_DefaultRole",
    VisibleToAllUsers=True,
    ReleaseLabel="emr-5.31.0", # Semantically greater than 5.7.0 and EMR actually supports CustomAmiId
    CustomAmiId="ami-1234567"
)

@mock_emr
def reproduce_issue():
    client = boto3.client("emr", region_name="us-east-1")
    client.run_job_flow(**run_job_flow_args)
```

**Comment by bblommers** [comment]
Thanks for raising this @PanJoseph! Marking it as a bug

**Comment by bblommers** [comment]
This is now fixed as of moto >= 1.3.16.dev138
